Doc/library/functools.rst

```diff
@@ -208,6 +208,55 @@ The :mod:`functools` module defines the following functions:
       Returning NotImplemented from the underlying comparison function for
       unrecognised types is now supported.
 
+.. function:: topsort(pairs)
+   Topologically sort a list of (parent, child) pairs. Return a sequence of
+   the elements in dependency order (parent to child order).
+
+   If a dependency cycle is found it raises :exc:`CycleError`.
```

Comment:
`CycleError` needs to be documented in the "Built-in Exceptions" chapter.

Reply:
Correct me if I am wrong but I don't think this needs to go into "Built-in Exceptions" chapter since this is custom exception specific to this function and there are a lot of other places (`CookieError`, `NetrcParseError`, etc.) that are not present in the chapter.